src/coreclr/vm/dispatchinfo.h

```diff
@@ -52,18 +52,47 @@ enum CultureAwareStates
 };
 
 // This structure represents a dispatch member.
-struct DispatchMemberInfo
+class DispatchMemberInfo
 {
-    DispatchMemberInfo(DispatchInfo *pDispInfo, DISPID DispID, SString& strName);
+private: // static
+    static MethodTable*     s_pMemberTypes[NUM_MEMBER_TYPES];
+    static EnumMemberTypes  s_memberTypes[NUM_MEMBER_TYPES];
+    static int              s_iNumMemberTypesKnown;
+
+public:
+    DispatchMemberInfo(DispatchInfo* pDispInfo, DISPID DispID, SString& strName);
     ~DispatchMemberInfo();
 
     // Helper method to ensure the entry is initialized.
     void EnsureInitialized();
 
-    BOOL IsNeutered()
+    DISPID GetDISPID()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_DispID;
+    }
+
+    DispatchMemberInfo* GetNext()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_pNext;
+    }
+
+    DispatchMemberInfo** GetNextField()
```

Comment:
A nit - the method name is confusing to me, would you mind renaming it to `GetNextPtr`? E.g. `ArrayListBase::BlockIterator` uses the same naming.

Reply:
Thank you for the suggestion. I didn't like it either. Prior art is always preferred in these cases.